pkg/workloadapi/workload.proto

```diff
@@ -215,6 +216,33 @@ enum TunnelProtocol {
   // Future options may include things like QUIC/HTTP3, etc.
 }
 
+// NativeTunnel describes how a workload (application or gateway)
+// can consume tunnel connections directly.
+message NativeTunnel {
+  // A target natively handles this type of traffic.  
+  GatewayProtocol protocol = 1;
+  // optional: if set, traffic should be sent to this port after the last zTunnel hop
+  uint32 port;
+}
+
+enum GatewayProtocol {
+  // Bytes are just copied directly.
+  NONE = 0;
+
+  // The workload natively supports HBONE
+  // For the zTunnel, this means:
+  // * Requests *from* this workload do not need to be tunneled if they already are tunneled by the tunnel_protocol.
+  // * Requests *to* this workload, via the tunnel_protocol, do not need to be de-tunneled.
+  HBONE = 1;
```

Comment:
And if we are adding - perhaps it's time to start using h2connect as protocol name ?